### PR TITLE
refactor(installer): remove CRDs from Helm chart, delegate to operator --install-crds

### PR DESCRIPTION
**Pull request by lexfrei**
## What this PR does

Removes CRDs from the cozy-installer Helm chart `crds/` directory and delegates
CRD lifecycle management entirely to the operator via `--install-crds` flag.

The operator already applies embedded CRDs via server-side apply on every startup,
making the Helm `crds/` directory redundant. Helm only installs CRDs on initial
`helm install` and never updates or deletes them on upgrade/uninstall, which causes
CRDs to become stale over time.

Changes:
- Remove `packages/core/installer/crds/` (Packages and PackageSources CRDs)
- Remove `templates/packagesource.yaml` Helm template — PackageSource is now
  created by the operator at startup using server-side apply
- Add `installPlatformPackageSource()` function to operator with SSA
- Move variant validation from deleted template to `cozystack-operator.yaml`
- Simplify `update-codegen.sh` to use single CRD destination
- Update Makefile to source CRDs from `internal/crdinstall/manifests/`
- Update E2E tests to wait for operator-managed CRDs and PackageSource
- Add unit tests for PackageSource creation

### Release note

```release-note
[installer] CRDs are no longer shipped in the Helm chart crds/ directory. The operator now manages CRD lifecycle via --install-crds flag, ensuring CRDs stay up to date on every startup.
```

## Summary by CodeRabbit

* **New Features**
  * Operator now installs CRDs on startup and auto-creates the platform PackageSource.

* **Improvements**
  * Installer waits for CRDs to be established and for the platform PackageSource to be present before proceeding.
  * Deployment variant selection now has stricter validation to prevent invalid choices.

* **Chores**
  * Cleaned up legacy CRD templates and updated build/install scripting paths.

* **Tests**
  * Added comprehensive tests covering platform PackageSource installation and URL/ref parsing.

### CODE REVIEW

File: cmd/cozystack-operator/main.go [hunk 1]
```diff
@@ -552,3 +577,108 @@ func generateGitRepository(name, repoURL string, refMap map[string]string) (*sou
...
+							Path: "core/platform",
+							Install: &cozyv1alpha1.ComponentInstall{
+								Namespace:   "cozy-system",
+								ReleaseName: "cozystack-platform",
+							},
+							ValuesFiles: []string{"values.yaml", "values-isp-full-generic.yaml"},
+						},
+					},
+				},
+			},
```
[the remark this answers is not in the document]
Comment (lexfrei) [reply cmd/cozystack-operator/main.go]: Addressed in d0a6ddd7 — replaced repetitive variant struct literals with a loop over variant data.

File: cmd/cozystack-operator/main.go [hunk 2]
```diff
@@ -224,6 +224,31 @@ func main() {
...
+		}
+		setupLog.Info("Creating platform PackageSource", "platformSourceName", platformSourceName)
+		psCtx, psCancel := context.WithTimeout(mgrCtx, 2*time.Minute)
+		defer psCancel()
+		if err := installPlatformPackageSource(psCtx, directClient, platformSourceName, sourceRefKind); err != nil {
+			setupLog.Error(err, "failed to create platform PackageSource")
+			os.Exit(1)
+		}
+		setupLog.Info("Platform PackageSource creation completed successfully")
+	}
```
[the remark this answers is not in the document]
Comment (lexfrei) [reply cmd/cozystack-operator/main.go]: Addressed in 4843a617 — added `platformSourceURL \!= ""` guard and comprehensive tests for edge cases.
